src/herder/HerderSCPDriver.cpp

```diff
@@ -1285,4 +1286,84 @@ HerderSCPDriver::TxSetValidityKeyHash::operator()(
     hashMix(res, std::get<3>(key));
     return res;
 }
+
+uint64
+HerderSCPDriver::getNodeWeight(NodeID const& nodeID, SCPQuorumSet const& qset,
+                               bool const isLocalNode) const
+{
+    Config const& cfg = mApp.getConfig();
+    if (protocolVersionIsBefore(mApp.getLedgerManager()
+                                    .getLastClosedLedgerHeader()
+                                    .header.ledgerVersion,
+                                APPLICATION_SPECIFIC_WEIGHT_PROTOCOL_VERSION) ||
+        !cfg.VALIDATOR_WEIGHT_CONFIG.has_value() ||
+        cfg.FORCE_APPLICATION_AGNOSTIC_WEIGHT_FUNCTION)
+    {
+        // Fall back on old weight algorithm if any of the following are true:
+        // 1. The network has not yet upgraded to
+        //    APPLICATION_SPECIFIC_WEIGHT_PROTOCOL_VERSION,
+        // 2. The node is using manual quorum set configuration, or
+        // 3. The node has the FORCE_APPLICATION_AGNOSTIC_WEIGHT_FUNCTION flag
+        //    set
+        return SCPDriver::getNodeWeight(nodeID, qset, isLocalNode);
+    }
+
+    ValidatorWeightConfig const& vwc =
+        mApp.getConfig().VALIDATOR_WEIGHT_CONFIG.value();
+
+    auto entryIt = vwc.mValidatorEntries.find(nodeID);
+    if (entryIt == vwc.mValidatorEntries.end())
+    {
+        // This shouldn't be possible as the validator entries should contain
+        // all validators in the config. For this to happen, `getNodeWeight`
+        // would have to be called with a non-validator `nodeID`. Throw if
+        // building tests, and otherwise fall back on the old algorithm.
+#ifdef BUILD_TESTS
+        throw std::runtime_error(
+            fmt::format(FMT_STRING("Validator entry not found for node {}"),
+                        toShortString(nodeID)));
+#endif
+        return SCPDriver::getNodeWeight(nodeID, qset, isLocalNode);
+    }
+
+    ValidatorEntry const& entry = entryIt->second;
+    if (entry.mQuality == ValidatorQuality::VALIDATOR_LOW_QUALITY)
+    {
+        // Special case: low quality validators have weight `0`
```

Comment:
I know this was in the original design, but I'm now wondering if there's a reason to not give a chance to LOW quality nodes at all?

Reply:
This allows for evaluation of new validators who will perform most SCP tasks except for choosing txsets.

CCing @MonsieurNicolas because this was originally his idea.

Reply:
yes exactly -- this allows an org to add as LOW validators to their own qset without risking their own reputation. We'll need to document this